src/rendering/vcPolygonModel.cpp

```diff
@@ -308,7 +308,7 @@ udResult vcPolygonModel_CreateFromOBJ(vcPolygonModel **ppPolygonModel, const cha
   {
     vcOBJ::Face *pFace = &pOBJReader->faces[f];
     if (pFace->mat == -1)
-      continue;
+      pFace->mat = 0;
```

Comment:
It looks to me like the vcOBJ_Load function doesn't set a default material at index 0, it might be prudent to define one, otherwise it's less clear that the model isn't likely supposed to look like that, so above where it says
``` // always have one
  if (pOBJReader->materials.length == 0)
  {
    vcOBJ::Material *pMat = pOBJReader->materials.PushBack();
    pMat->Ka = udFloat3::create(1.f);
    pMat->Kd = udFloat3::create(1.f);
  }```
Maybe push that back regardless of the condition and here set pFace->mat to ->materials.length - 1

Reply:
Or I could be concerning myself with an improper use of the obj format, idk

Reply:
Could work yeah. I'd rather change as little as possible though.